src/sentry/workflow_engine/endpoints/validators/base/detector.py

```diff
@@ -65,8 +65,8 @@ def data_conditions(self) -> BaseDataConditionValidator:
 
     def update(self, instance: Detector, validated_data: dict[str, Any]):
         with transaction.atomic(router.db_for_write(Detector)):
-            instance.name = validated_data.get("name", instance.name)
-            instance.type = validated_data.get("detector_type", instance.group_type).slug
```

Comment:
I removed the `type` line because I don't think we want to support changing a detector's type